--- a/src/echo.c
+++ b/src/echo.c
@@ -17,9 +17,7 @@
 #include <config.h>
 #include <stdio.h>
 #include <sys/types.h>
-#include "system.h"
-#include "assure.h"
-#include "c-ctype.h"
+#include "silt-system.h"
 
 /* The official name of this program (e.g., no 'g' prefix).  */
 #define PROGRAM_NAME "echo"
--- a/src/basename.c
+++ b/src/basename.c
@@ -19,7 +19,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 
-#include "system.h"
+#include "silt-system.h"
 #include "quote.h"
 
 /* The official name of this program (e.g., no 'g' prefix).  */
--- a/src/dirname.c
+++ b/src/dirname.c
@@ -22,7 +22,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 
-#include "system.h"
+#include "silt-system.h"
 
 /* The official name of this program (e.g., no 'g' prefix).  */
 #define PROGRAM_NAME "dirname"
